Module-11/Library/Repositories/FileSystemDocumentRepository.cs

```diff
@@ -0,0 +1,64 @@
+﻿using Library.Documents;
+using System.Runtime.Caching;
+using System.Text.Json;
+using Microsoft.Extensions.Configuration;
+
+namespace Library.Repositories;
+
+public class FileSystemDocumentRepository : IDocumentRepository
+{
+    private readonly MemoryCache _cache = MemoryCache.Default;
+    private readonly Dictionary<string, TimeSpan?>? _cacheDocumentExpirationTimeSpans;
+    
+    public FileSystemDocumentRepository(IConfigurationRoot config)
+    {
+        _cacheDocumentExpirationTimeSpans = config.GetSection("DocumentCacheExpirationTimeSpans")
+            .Get<Dictionary<string, TimeSpan?>>();
+    }
+
+    private const string StoragePath = "documents/";
+    public IEnumerable<IDocument?> GetByNumber(int documentNumber)
+    {
+        var matchingFileNames = Directory.GetFiles(StoragePath, $"*_{documentNumber}.json");
+        if (matchingFileNames.Length == 0)
+        {
+            return Enumerable.Empty<IDocument>();
+        }
+        var types = AppDomain.CurrentDomain
+            .GetAssemblies()
+            .SelectMany(x => x.GetTypes())
+            .ToList();
+
+        return matchingFileNames.Select(fileName =>
+        {
+            var documentTypeName = Path.GetFileNameWithoutExtension(fileName).Split('_')[0];
+            var type = types.FirstOrDefault(t => t.Name == documentTypeName);
+
+            var json = GetCardJson(fileName, documentTypeName);
```

Comment:
I don't like GetCardJson method because there is no guarantee that it will actually return a valid json.
Try to think of a way to reorganize deserialization logic